lib/react_client.dart

```diff
@@ -846,6 +846,12 @@ final Expando<Function> _originalEventHandlers = new Expando();
 /// Dart synthetic counterparts.
 Map unconvertJsProps(/* ReactElement|ReactComponent */ instance) {
   var props = Map.from(JsBackedMap.backedBy(instance.props));
+
+  // Catch if a Dart component has been passed in.
+  if (props['internal'] is ReactDartComponentInternal || (props['style'] != null && props['style'] is Map)) {
```

Comment:
Pardon my ignorance - but I'm not following why a `Map` value for `props['style']` indicates that a Dart component instance has been passed in here?  Possibly deserves a bit more info in the code comment, IMO.

Reply:
Good call. I added a comment explaining my logic. Let me know if there's a better way to detect `Component2` - it's not the best to rely on the style prop, but it does stop the error that created the need for the PR.